### PR TITLE
Compatibility update3

### PR DESCRIPTION
**Pull request by lorforlinux**
@RobertCNelson @jadonk please review my code, Everything has been tested on BBBWL and BBAI. 

This PR provides BBAI support for 4.3" LCD cape. It also includes a compatible Device Tree Overlay for CTAG face2|4 multichannel soundcard - 8 audio channel, thanks to [NiklasWan](https://github.com/NiklasWan) for helping with that.

### CODE REVIEW

File: src/arm/bbai-bone-buses.dtsi [hunk 1]
```diff
@@ -528,7 +528,7 @@ bone_eqep_2: &eqep1{
 
 // Bone sound
 &ocp {
-	bone_sound {
+	bone_sound: bone_sound {
```
Comment (jadonk) [inline src/arm/bbai-bone-buses.dtsi:538]: I'm going through these commit-by-commit so I can best understand what is going on. I will be looking for a future commit that merges these two files into one to know that the compatibility is working. It might have been nice to see changes to only one file, renaming it to a common-name and elimination of the other file.

File: src/arm/bbai-bone-buses.dtsi [hunk 2]
```diff
@@ -468,8 +468,15 @@ bone_timer_5: &timer15 {
 
 // McASP
 bone_mcasp_0: &mcasp1 {
+        #sound-dai-cells= <0>;
```
Comment (jadonk) [inline src/arm/bbai-bone-buses.dtsi]: Indentation looks inconsistent. Let's try using tabs. If you need to do a formatting fix, do that separate from functional fixes.

File: src/arm/bbai-bone-buses.dtsi [hunk 3]
```diff
@@ -468,8 +468,15 @@ bone_timer_5: &timer15 {
...
 	assigned-clocks = <&ipu_clkctrl DRA7_MCASP1_CLKCTRL 24>;
 	assigned-clock-parents = <&sys_clkin2>; /* 22579200 Hz (see dra7xx-clocks.dtsi) */
+        serial-dir = <  /* 0: INACTIVE, 1: TX, 2: RX */
```
Comment (jadonk) [inline src/arm/bbai-bone-buses.dtsi:474]: These files are likely to be referenced a lot. Is there a reference to what each of these numbers refer? I knew it at one point, but I forget so easily. Just don't want blind copy-and-paste.

File: src/arm/bbb-bone-buses.dtsi [hunk 4]
```diff
@@ -485,7 +485,12 @@ bone_timer_5: &not_available {
 
 // McASP
 bone_mcasp_0: &mcasp0 {
-
+        serial-dir = <  /* 0: INACTIVE, 1: TX, 2: RX */
+		2 0 1 0
```
Comment (jadonk) [inline src/arm/bbb-bone-buses.dtsi:489]: Interesting that this is different. Is it possible to instead put the I2S (I think this is I2S) in as a peripheral defined as part of the main board (bone bus) device tree and simply add target device specific (not bus pinout specific) settings in this file?

Comment (jadonk) [reply src/arm/bbb-bone-buses.dtsi:489]: Oh, nevermind. That is exactly what you are doing. I completely lost the fact that bbb-bone-buses and bbai-bone-buses *ARE* the bone bus definitions and *NOT* the overlays. My bad on silly comments.

File: src/arm/bbai-bone-buses.dtsi (hunk 1 above)
Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi:538]: Bad comment. Didn't realize this *WAS* -bone-buses.dtsi.

Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi:538]: Note, please look at all places I2S could be routed out. I think thinking of this more as an I2S bus, rather than "sound" can be useful to making in generic. This is a new bus/interface like SPI, I2C, UART, etc. and should be treated the same in the cape compatibility documentation and definition.

Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi:538]: For example, I think it should likely be given an index.

File: src/arm/am5729-beagleboneai.dts [hunk 5]
```diff
@@ -23,7 +23,7 @@
 	aliases {
 		rtc0 = &tps659038_rtc;
 		rtc1 = &rtc;
-		display0 = &hdmi_conn;
+		display1 = &hdmi_conn;
```
Comment (jadonk) [inline src/arm/am5729-beagleboneai.dts:26]: Can you say why this change was needed and what it means?

File: src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts [hunk 6]
```diff
@@ -0,0 +1,173 @@
+/*
```
Comment (jadonk) [inline src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts]: I'll be looking for where the secondary copy of this is deleted and a common file is used. It would be easier for me to follow if that was the first step.

File: src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+				data-lines = <16>;
+				remote-endpoint = <&lcd_in>;
+			};
+		};
+	};
+};
+
+&{/} {
+	aliases {
+		display0 = &lcd0;
```
Comment (jadonk) [inline src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts]: Is it desired to make the LCD the primary display and the HDMI the secondary? Is that what this is doing?

File: src/arm/overlays/BBAI-4D4C-00A1.dts [hunk 8]
```diff
@@ -15,7 +15,7 @@
  
 &{/chosen} {
 	overlays {
-		BBAI-BB-BONE-4D4C-01-00A1 = __TIMESTAMP__;
+		BBAI-4D4C-00A1 = __TIMESTAMP__;
```
Comment (jadonk) [inline src/arm/overlays/BBAI-4D4C-00A1.dts:18]: Shouldn't this be non-BeagleBone-AI specific?

File: src/arm/bbai-bone-buses.dtsi [hunk 9]
```diff
@@ -468,7 +468,7 @@ bone_timer_5: &timer15 {
 
 // McASP
 bone_mcasp_0: &mcasp1 {
-        #sound-dai-cells= <0>;
+    #sound-dai-cells= <0>;
```
Comment (jadonk) [inline src/arm/bbai-bone-buses.dtsi:471]: Can you fix the tabs across the document to make these things line up?

File: src/arm/overlays/BBAI-4D4C-00A1.dts [hunk 10]
```diff
@@ -0,0 +1,173 @@
+/*
```
Comment (jadonk) [inline src/arm/overlays/BBAI-4D4C-00A1.dts:1]: Does this work on BeagleBone Black now? Can you rename as appropriate?

**Review by jadonk — CHANGES_REQUESTED** [review]
* Rename BBAI-4D4C-00A1.dts and make this common across BeagleBone Black and BeagleBone AI.
* Add requested comments and clean-up formatting.

File: src/arm/bbai-bone-buses.dtsi (hunk 3 above)
Comment (lorforlinux) [reply src/arm/bbai-bone-buses.dtsi:474]: I have no idea about these indexes, Niklas fixed these as you can see this contribution comes from him.

File: src/arm/bbai-bone-buses.dtsi (hunk 2 above)
Comment (lorforlinux) [reply src/arm/bbai-bone-buses.dtsi]: Indentation problem again 😂 I always try my best to resolve them but one always slips in.

File: src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts (hunk 7 above)
Comment (lorforlinux) [reply src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts]: AM5729 SDK follows this for the LCD overlays and not make things complicated I opted to do the same. I don't think by creating the alias we are setting the priority of the video output devices, are we @RobertCNelson?

File: src/arm/overlays/BBAI-4D4C-00A1.dts (hunk 8 above)
Comment (lorforlinux) [reply src/arm/overlays/BBAI-4D4C-00A1.dts:18]: Sorry this one only works for BBAI, but we should merge this. Even when I create a compatible overlay this overlay will be the go to resource for new LCD cape overlays and will help to understand what's going on inside compatible LCD overlays because things from this will go to several files when compatibility layer will support LCDs.

File: src/arm/bbai-bone-buses.dtsi (hunk 9 above)
Comment (lorforlinux) [reply src/arm/bbai-bone-buses.dtsi:471]: Believe me, i am trying my best with tabs. Things look pretty here and becomes ugly when pushed 😞

File: src/arm/overlays/BBAI-4D4C-00A1.dts (hunk 10 above)
Comment (lorforlinux) [reply src/arm/overlays/BBAI-4D4C-00A1.dts:1]: Nope, this only works on BBAI!

**Comment by lorforlinux** [comment]
@jadonk I suppose there is nothing I can improve on this PR, I will try to create a compatible LCD overlay for 4.3" cape and ship it with compatibility update 4. The sound (I2S) stuff I got working with generous help from Niklas and I really don't want to break it by updating things until I have the setup to test what I am doing which I don't have at this moment :(

File: src/arm/bbai-bone-buses.dtsi (hunk 1 above)
Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi:538]: The I2S port is indexed, why not the audio port?

File: src/arm/bbai-bone-buses.dtsi (hunk 3 above)
Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi]: Please discover and document.

https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/sound/davinci-mcasp-audio.txt

That provides a start, but it probably needs to be patched with something that indicates the I/O pin indexes. The point of the cape compatibility layer is to make it clear to people how they can create hardware using the interfaces that "just works" with the software. At least knowing what might or might not ever need to change would help.

File: src/arm/am5729-beagleboneai.dts (hunk 5 above)
Comment (jadonk) [reply src/arm/am5729-beagleboneai.dts:26]: ping.

File: src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts (hunk 6 above)
Comment (jadonk) [reply src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts]: OK, I think we are still stuck on getting this into one file? I think making it clear in the commit log where the hold-up is would go a long way in enabling someone else to pick it up.

File: src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts (hunk 7 above)
Comment (jadonk) [reply src/arm/overlays/BBAI-BB-BONE-4D4C-01-00A1.dts]: @RobertCNelson ?? Do you know who we can ask?

File: src/arm/overlays/BBAI-4D4C-00A1.dts (hunk 8 above)
Comment (jadonk) [reply src/arm/overlays/BBAI-4D4C-00A1.dts:18]: Agree we should merge, but please document the current limitations so it can be picked up in the future.

File: src/arm/bbai-bone-buses.dtsi (hunk 9 above)
Comment (jadonk) [reply src/arm/bbai-bone-buses.dtsi]: There are a number of scripts you could run to format it after. I want to avoid it becoming a bigger mess.

File: src/arm/overlays/BBAI-4D4C-00A1.dts (hunk 10 above)
Comment (jadonk) [reply src/arm/overlays/BBAI-4D4C-00A1.dts:1]: Please note what deltas we cannot resolve.

**Comment by jadonk** [comment]
Please add the requested comments at least. We'll want someone to pick up where you left off. Thanks again for taking the extra step to wrap this up and get it working.

**Comment by lorforlinux** [comment]
@jadonk @RobertCNelson Can anyone please add `hacktoberfest-accepted` label to this PR?
